File: spring-data/spring-data-jpa/jpa-hsqldb/src/main/java/com/example/ApplicationEventListener.java
package com.example;

import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Component
@RequiredArgsConstructor
@Slf4j
public class ApplicationEventListener {
    private final UserRepository userRepository;

    @EventListener(ApplicationReadyEvent.class)
    public void readyEvent() {
        User user = new User();
        user.setName("user6");
        userRepository.save(user);

        userRepository.findAll()
                      .forEach(entity -> log.info("{}", entity));
    }
}

File: jdbc/spring-jdbc-kotlin/src/main/kotlin/com/example/ApplicationEventListener.kt
package com.example

import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class ApplicationEventListener(private val userRepository: UserRepository) {
    @EventListener(ApplicationReadyEvent::class)
    fun readyEvent() {
        userRepository.findAll().forEach {
            println(it)
        }
    }
}
